libdrgn/python/type.c

```diff
@@ -1578,13 +1582,16 @@ static DrgnType *compound_type(PyObject *tag_obj, PyObject *size_obj,
 		if (!cached_members_obj)
 			return NULL;
 		num_members = PyTuple_GET_SIZE(cached_members_obj);
+		members = malloc(sizeof(struct drgn_type_member) * num_members);
+		if (!members)
+			goto err;
 
 		type_obj = DrgnType_new(qualifiers, num_members,
 					sizeof(struct drgn_type_member));
```

Comment:
This should be `DrgnType_new(qualifiers, 0, 0)` now, since the members array isn't embedded anymore.

Reply:
I decided to do this on the leading 3 commits, and in the cleanup commit I just removed those parameters entirely (as they weren't used).